src/User32.Desktop/User32+DESKTOP_ACCESS_MASK.cs

```diff
@@ -0,0 +1,271 @@
+            [Flags]
+            public enum SpecificRight : uint
+            {
+                /// <summary>
+                /// The bit mask that covers all possible access rights for the desktop.
+                /// </summary>
+                DESKTOP_ALL_ACCESS = 0x000001FF,
+
+                /// <summary>
+                /// Required to create a menu on the desktop.
+                /// </summary>
+                DESKTOP_CREATEMENU = 0x00000004,
+
+                /// <summary>
+                /// Required for the desktop to be enumerated.
+                /// </summary>
+                DESKTOP_ENUMERATE = 0x00000040,
+
+                /// <summary>
+                /// Required to establish any of the window hooks.
+                /// </summary>
+                DESKTOP_HOOKCONTROL = 0x00000008,
+
+                /// <summary>
+                /// Required to perform journal playback on a desktop.
+                /// </summary>
+                DESKTOP_JOURNALPLAYBACK = 0x00000020,
+
+                /// <summary>
+                /// Required to perform journal recording on a desktop.
+                /// </summary>
+                DESKTOP_JOURNALRECORD = 0x00000010,
+
+                /// <summary>
+                /// Required to read objects on the desktop.
+                /// </summary>
+                DESKTOP_READOBJECTS = 0x00000001,
+
+                /// <summary>
+                /// Required to create a window on the desktop.
+                /// </summary>
+                DESKTOP_CREATEWINDOW = 0x00000002,
+
+                /// <summary>
+                /// Required to activate the desktop using the <see cref="SwitchDesktop"/> function.
+                /// </summary>
+                DESKTOP_SWITCHDESKTOP = 0x00000100,
+
+                /// <summary>
+                /// Required to write objects on the desktop.
+                /// </summary>
+                DESKTOP_WRITEOBJECTS = 0x00000080
+            }
+
+            /// <summary>
+            /// Gets the DESKTOP_ACCESS_MASK as a 32-bit unsigned integer.
+            /// </summary>
+            public uint Value { get; }
+
+            /// <summary>
+            /// Gets the DESKTOP_ACCESS_MASK as a 32-bit signed integer.
+            /// </summary>
+            [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+            public int AsInt32 => (int)this.Value;
+
+            /// <summary>
+            /// Gets the basic generic rights of this value, the ones that Windows can set and read for any object
+            /// </summary>
+            public Kernel32.ACCESS_MASK.GenericRight BasicGenericRights => (Kernel32.ACCESS_MASK.GenericRight)(this.Value & GenericRightsMask);
+
+            /// <summary>
+            /// Gets the generic rights of this value for interactive desktops.
+            /// </summary>
+            public GenericRight InteractiveWindowStationGenericRights => (GenericRight)(this.Value & GenericRightsMask);
```

Comment:
Where does the term "InteractiveWindowStation" come from? That doesn't seem to align with the Desktop term you're using elsewhere. Should this be called `GenericRights`?